src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -746,8 +746,11 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
   start <- max(1, grep(.rs.reRmdChunkBegin(), code, perl = TRUE))
   end   <- min(length(code), grep(.rs.reRmdChunkEnd(), code, perl = TRUE))
 
-  paste(code[(start + 1):(end - 1)], collapse = "\n")
-});
+  code <- code[(start + 1):(end - 1)]
+  code <- code[!grepl("^\\s*#\\|", code) & code != ""]
```

Comment:
I'm not sure if we want to remove empty lines; it might cause issues with diagnostic markers in R code chunks (since we rely on line numbers produced from the errors to figure out where to place e.g. error markers in the gutter)

This is worth testing + verifying though. If this is an issue, I think we could replace the Quarto comments with empty lines.

Reply:
I might be thinking of document renders though, where this code won't be relevant... so this code should probably still be fine, but I still think it's worth a sanity check?

Reply:
I'll check if e.g. `sql` is fine with an empty line

Reply:
That seems to work so I've updated.